### PR TITLE
Group by string value in StringColumn

### LINKED ISSUE #192: Possibility to group by string value in StringColumn
Currently, it is not possible to group a `StringColumn` by the actual value of the string (only `starts with ...` and `RegExp ...` are supported). 
With identical strings, it sometimes does not make sense to make it a `CategoricalColumn`, but grouping should be possible nonetheless.

### PR DESCRIPTION
**Pull request by dvmichaelpuehringer**
closes #192  

**prerequisites**: 
 * [x] branch is up-to-date with the branch to be merged with, i.e. develop
 * [x] build is successful
 * [x] code is cleaned up and formatted 
 * [x] tested with Firefox 52, Firefox 57+, Chrome 64+, MS Edge 16+

### Summary
This PR adds the possibility of grouping a `StringColumn` by its actual value. It adds the option to the `groupString` dialog extension and adds a new type `EStringGroupCriteriaType` and interface `IStringGroupCriteria` to the column, allowing for more than just two criteria types (`string | RegExp` as before).

Peek 2019-10-10 16-16

### CODE REVIEW

**Comment by mstreit** [comment]
Very useful feature!

**Comment by thinkh** [comment]
@dvmichaelpuehringer I'd further suggest to use the "standard" dialog buttons (e.g., in the rename dialog):

grafik

That would make UI a bit more consistent.

**Comment by thinkh** [comment]
I agreed with @dvmichaelpuehringer that we will change the button UI in a different PR.

File: demo/string_grouping.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+<script>
+  window.onload = function () {
+    const arr = [];
+    for (let i = 0; i < 1000; ++i) {
+      arr.push({
+        a: Math.random() * 10,
+        d: 'Row ' + i,
+        group: Math.floor(Math.random() * 2000).toString()
+      })
```
Comment (thinkh) [inline demo/string_grouping.html]: @dvmichaelpuehringer For me the demo does not show duplicate rows that I could group. Can you please fix it?

Comment (dvmichaelpuehringer) [reply demo/string_grouping.html]: Should be fixed with https://github.com/lineupjs/lineupjs/pull/193/commits/a0df751d459915a0969fa05814f114a650ffc3fa

**Pull request merged**